### DISCUSSION TITLE
Avoid Airflow dag from going into running state if previous schedule is in failed.

### DISCUSSION THREAD

**Discussion by Raul824** [apache/airflow#29277] [Q&A]
Hello All,

I have multiple dags for daily schedule. Each dag has 3+ tasks. max_active_run is set to 1 and depends on past is true.
I don't want to run the new dag until previous finishes.
So if in a scenario a dag is scheduled for 27,28,29. And below process happens.
1. Jan 27th dag 2nd task fails this puts dag in failed status.
2. As max_active_run is 1 and failed isn't considered active 28th Dag goes into running.
3. No tasks start as depends on past is true.
4. Now I want to rerun 2nd task of 27th but it goes in queue because max_active_run is 1 
5. Now we have to mark 28th running dag in failed
6. As soon as we set 28th running dag in failed 29th goes in running.
7. than after setting all the queued dag to failed we can rerun 27th 2nd task.

Is there any way to keep the next dag schedule to not go in running if the current schedule is in failed.
Or is there any better solution to achieve this.

**Comment by hussein-awala** [comment]
I have the same problem on some of my dags, I didn't find a solution for that.

For that, I'm working on a PR which add a new param `depends_on_past` for the dag, if it's `True`, the scheduler will wait the DagRun to pass in `success` or `skipped` state before creating a new DagRun, without the need to specify `max_active_run`, in this case we can keep set it to a value > 1 in order to run some backfill in the same time.
